School/views.py

```diff
@@ -211,11 +212,57 @@ def lesson_detail(request, lesson_id):
     return render (request, 'lesson.html', ctx)
 
 
+def countavgmark(activities):
+    all_marks = []
+
+    for s in activities:
+        marks = s.mark.number
+        all_marks.append(marks)
+
+    start = 0
+
+    for m in all_marks:
+        start = start + m
+
+    if all_marks:
+        x = Decimal(float(start) / len(all_marks))
+        avg = round(x,2)
+    else:
+        avg = 0
+
+    return avg
+
+
+def count_avg_mark_discpline(activities_for_marksd):
```

Comment:
Will it work for any activities queryset? Or is it specific to activities filtered only by discipline ?